### ISSUE TITLE
login.chi returns EMPTY_RESPONSE

### ISSUE THREAD

**Issue by jpaulovic** [coova/coova-chilli#118]
Hi, I have installed CoovaChilli and radius but when I try to login with admin account to get to configuration page, the www/login.chi returns EMPTY_RESPONSE after clicking Login & Accept Terms. 
I can't seem to figure out why this is happening, after looking into the login.chi script I would expect to get either login success or failure page. Please can You help me fix this issue?

**Comment by sevan** [comment]
Can you try running from the very latest version from git repo & confirm if anything has changed

**Comment by sevan** [comment]
@jpaulovic I've ammended the README.md file in the repo, if you pull from github again, you'll find the instructions needed to get started. 
You need to have autoconf, automake, libtool & gengetopt installed.

**Comment by jpaulovic** [comment]
I've purged the 1.3.0 version from the system via dpkg, then compiled and installed latest version from git repo via the recommended metod (./configure without parameters, make, make install) and now I immediately get segmentation fault when I try to run chilli, like this:
root@raspberrypi:# chilli --fg --debug
Segmentation fault
root@raspberrypi:#

**Comment by sevan** [comment]
What distro are you running on your pi?

**Comment by jpaulovic** [comment]
I'm running Raspbian GNU/Linux 7 (wheezy)

**Comment by sevan** [comment]
is there a .core file generated after it segfaults?

**Comment by jpaulovic** [comment]
Seems there isn't:
root@raspberrypi:/# chilli --fg --debug
Segmentation fault
root@raspberrypi:/# find / -name *.core
root@raspberrypi:/#

**Comment by sevan** [comment]
You've ditched any trace of previous config files from /etc?

**Comment by jpaulovic** [comment]
No, I left config files there so I don't have to configure it again. However, it at least started in 1.3.0. Do I have to delete something from /etc/chilli?

**Comment by sevan** [comment]
try starting with a clean state just to rule out your config.

**Comment by nzamps** [comment]
...or run it within gdb (if you can add that):  gdb --args chilli --fg --debug  then enter bt when it segfaults.

**Comment by jpaulovic** [comment]
gdb:
Starting program: /usr/local/sbin/chilli --fg --debug
warning: Could not load shared library symbols for linux-vdso.so.1.
Do you need "set solib-search-path" or "set sysroot"?
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".

Program received signal SIGILL, Illegal instruction.
0x76c73608 in ?? () from /usr/lib/arm-linux-gnueabihf/libcrypto.so.1.0.0
--

Will try to remove the config and try with clean installation tomorrow.

**Comment by spirou1997** [comment]
Hi,

I've the same problem.
After login, i'm redirected and login.chi returns EMPTY_RESPONSE
Is your problem solved ? If yes, how ?
Thank you

**Comment by MrDeviance** [comment]
Hi,

I've the same problem too....
How could you solve this problem? 
We wait your answer!:)) 
Thanks!

**Comment by spirou1997** [comment]
Hi,

I solved it by changing path in /etc/chilli/wwsh and reinstalling haserl
I hope it work for you
Keep me informed

**Comment by MrDeviance** [comment]
It's work now!!!! Thank you!!!!! :dancer:

**Comment by jpaulovic** [comment]
Hi, sorry I didn't reply for such a long time, I didn't have time to test it...
Now when I tried to rebuild and install from latest sources, I get an error:
coova-chilli[13695]: (Re)processing options [/usr/local/var/run/chilli.13695.cfg.bin]
coova-chilli[13696]: running chilli_opt on /usr/local/var/run/chilli.13695.cfg.bin
coova-chilli: Error opening configuration file '/usr/local/etc/chilli/main.conf'
coova-chilli: Error opening configuration file '/usr/local/etc/chilli/hs.conf'
coova-chilli: Error opening configuration file '/usr/local/etc/chilli/local.conf'
coova-chilli[13695]: could not generate configuration (/usr/local/var/run/chilli.13695.cfg.bin), sleeping one second

Please can You give me some simple steb-by-step guide to make it work? I really don't understand what's going on unfortunately. Chilli put conf files to /usr/local/etc/chilli instead of /etc/chilli so I created a symlink, but now I get the error described above. I thought that those conf files should be created automatically when starting chilli, no? Also, I don't have anything related to chilli in /etc/init.d, however there is a chilli executable in /usr/local/etc/init.d/ but when I try to run it, I get an error
/usr/local/etc/init.d/chilli: 9: .: Can't open /etc/rc.d/init.d/functions

I don't have a folder rc.d in /etc/

**Comment by 6-break** [comment]
I could avoid this kind of errors to use https://github.com/coova/coova-chilli/issues/21 instead of using hasearl.

**Comment by diegohdk** [comment]
I'm having exactly the same problem. Already reinstalled haserl but it's not working.

@spirou1997 what path did you change?

**Comment by diegohdk** [comment]
Running gdb the output is:

```
coova-chilli[3603]: Serving file login.chi
coova-chilli[3603]: Running: /usr/local/etc/chilli/wwwsh /usr/local/etc/chilli/www/login.chi
Segmentation fault

Catchpoint 5 (signal SIGCHLD), 0x76cdea3c in select () at ../sysdeps/unix/syscall-template.S:81
81      ../sysdeps/unix/syscall-template.S: No such file or directory.
```

Is it some compilation error? Any library missing?

**Comment by jsubirat** [comment]
Hi all,

In case it helps, I managed to solve it based on spirou1997 comment (thanks!). The version that ships with Raspbian of haserl is (installed via apt-get) is the 0.9.33. However, I followed this tutorial http://www.pihomeserver.fr/en/2015/08/05/raspberry-pi-coovachilli-et-freeradius-pour-un-hotspot-wifi-avec-portail-captif/ to install the 0.9.35. I needed to install the 0.9.33 because if I installed the 0.9.35 from sources, my dpkg didn't detect that I had installed haserl, and didn't allow me to install coova-chilli. So, what I did was:

1) Install haserl 0.9.33 from repositories. It gets installed in /usr/bin/haserl
2) Install coova-chilli from repositories.
3) Install haserl 0.9.35 from sources (see tuto above). It gets installed in /usr/local/bin/haserl
4) Edit /etc/chilli/wwwsh and change the line with the haserl executable (I commented it, in fact), to point to the 0.9.35:
#haserl=$(which haserl 2>/dev/null)
haserl=/usr/local/bin/haserl

With this, I had no further Segmentation faults and worked correctly. I hope this helps!

**Comment by pihomeserver** [comment]
@jsubirat you can find a script to install the full portal here [pihomeserver/Pi-Hotspot-Script](https://github.com/pihomeserver/Pi-Hotspot-Script)
